OutOfSchool/OutOfSchool.WebApi/Services/Database/WorkshopService.cs

```diff
@@ -109,7 +105,10 @@ public async Task<WorkshopCreationResultDto> CreateV2(WorkshopDTO dto)
             // In case if DirectionId and DepartmentId does not match ClassId
             await FillDirectionsFields(dto).ConfigureAwait(false);
 
-            Func<Task<Workshop>> operation = async () =>
+            MultipleImageUploadingResult uploadingResult = null;
```

Comment:
(Another styling consideration, not a change request, just thing that irritates my eye, but I'm not insisting)
Variables `uploadingResult` and `uploadingCoverImageResult` are both results of `TransactionOperation` (latter name btw could be more specific - like `CreateWorkshopAndDependencies`) execution.
They are not set before it's execution (so they are not hidden parameters).
Why not make `TransactionOperation` more `pure` function ( [wiki](https://en.wikipedia.org/wiki/Pure_function) ) and return all results as, well, function result? Something like
`
async Task<(Workshop createdWorkshop, MultipleImageUploadingResult imagesUploadResult, Result<string> coverImageUploadResult)> TransactionOperation()
`
and then use it as
`
var (newWorkshop, imagesUploadResult, coverImageUploadResult) = await workshopRepository.RunInTransaction(TransactionOperation).ConfigureAwait(false);
`

Reply:
I see, and agree, my comment is irrelevant. 

In general I'll stand for less code but more diversity when possible - so `option 1` looks better.

Downsides of it are:
- possible necessity to explicitly specify method type - need to check how compiler wants it to be used, 
- and this is not backward compatible change - check please scope of changes (how many places are calling this `RunInTransaction<T>(Func<T> func)` from `EntityBaseRepository<TEntity>`, and should any additional code fix be done). 

If changes are way too big or syntax is too awkward - then `option 2`.

And ofc, I don't insist to do it in this PR.